joint_symbolic.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import itertools
+from scipy.signal import find_peaks
+
+plt.style.use('seaborn-v0_8-ticks')
+
+
+def get_peaks(signal):
+    peaks, _ = find_peaks(signal, distance=115)
+    difference = np.diff(peaks)
+    return peaks, signal[peaks], difference
+
+
+def make_empty_matrix():
+    my_permutations = [''.join(map(str, triplet)) for triplet in list((itertools.product([1, 0], repeat=3)))]
+    my_matrix = pd.DataFrame()
+    for triplet_1 in my_permutations:
+        for triplet_2 in my_permutations:
+            my_matrix.at[
+                "'" + triplet_1, "'" + triplet_2] = 0  # dash needed so that R does not remove the initial zeros
+    return my_matrix
+
+
+def get_diff_df(dataframe, is_fv, is_left, is_ekg):
+    sap = get_peaks(dataframe['abp_arm'])[1]
+    if is_fv:
+        sap = get_peaks(dataframe['fv_l'])[1]
+        if not is_left:
+            sap = get_peaks(dataframe['fv_r'])[1]
+
+    rr = get_peaks(dataframe['abp_arm'])[2] * 0.005  # in those without an ECG signal, we use ABP
+    if is_ekg:
+        rr = get_peaks(dataframe['ekg'])[2] * 0.005
```

Comment:
0.005 to stała wynikajaca z naszego próbkowania (200 Hz) - to tzreba opisac albo nie wrzucac